### PR TITLE
Added better test cases for hooks

### PR DESCRIPTION
**Pull request by aadilhasan**

### CODE REVIEW

File: src/lib/hooks/__tests__/useLazyQuery.test.tsx [hunk 1]
```diff
@@ -1,84 +1,107 @@
...
-import { useLazyQuery } from "../useQuery";
+import {
+  render,
+  act,
+  screen,
+  renderHook,
+  waitFor,
+} from "@testing-library/react";
+import { useLazyQuery, useQuery } from "../useQuery";
 import { init } from "../../init";
```
Comment (kaushalmeena) [inline src/lib/hooks/__tests__/useLazyQuery.test.tsx:10]: should remove unused imports

File: src/lib/hooks/__tests__/useLazyQueryWithPagination.test.tsx [hunk 2]
```diff
@@ -1,143 +1,366 @@
...
 
-    await act(async () => {
-      getByTestId("fetch-button").click();
+    it("should return formatted data if dataFormatter callback is provided", async () => {
+      const mockCompleteCallback = vi.fn();
+      const mockFormatterCallback = vi.fn();
+      const mockFormattedDataString = "mockFormattedDataString";
+
+      const { result } = renderHook(() =>
+        useLazyQueryWithPagination<"string">(testQuery, testVariables, {
```
Comment (kaushalmeena) [inline src/lib/hooks/__tests__/useLazyQueryWithPagination.test.tsx:167]: is there any significance of passing string type in useLazyQuery<"string"> here?

**Review by kaushalmeena — APPROVED** [review]

**Comment by kaushalmeena** [comment]
Just for future reference, should we include test for typescript types? using vitest we can test that also 
https://vitest.dev/guide/testing-types.html#:~:text=Vitest%20allows%20you%20to%20write,can%20change%20it%20with%20typecheck.

File: src/lib/hooks/__tests__/useLazyQueryWithPagination.test.tsx (hunk 2 above)
Comment (aadilhasan) [reply src/lib/hooks/__tests__/useLazyQueryWithPagination.test.tsx:167]: not much, added it just to test the generics.

File: src/lib/hooks/__tests__/useLazyQuery.test.tsx [hunk 3]
```diff
@@ -1,84 +1,107 @@
...
 
-    await act(async () => {
-      getByTestId("fetch-button").click();
+    it("should return formatted data if dataFormatter callback is provided", async () => {
+      const mockCompleteCallback = vi.fn();
+      const mockFormatterCallback = vi.fn();
+      const mockFormattedDataString = "mockFormattedDataString";
+
+      const { result } = renderHook(() =>
+        useLazyQuery<"string">(testQuery, testVariables, {
```
[the remark this answers is not in the document]
Comment (aadilhasan) [reply src/lib/hooks/__tests__/useLazyQuery.test.tsx:88]: not much, added it just to test the generics.

**Pull request merged**